pkg/discovery-api/discovery_api.go

```diff
@@ -0,0 +1,180 @@
+	cl := &DiscoveryClient{
+		Instance: instance,
+	}
+
+	var err error
+	cl.restConfig, err = NewRestClientConfig(rest.InClusterConfig)
+	if err != nil {
+		return nil
+	}
+
+	if cl.DiscoveryClient, err = discovery.NewDiscoveryClientForConfig(cl.restConfig); err != nil {
+		return nil
+	}
+
+	cl.ClientSet, err = dynamic.NewForConfig(cl.restConfig)
+	if err != nil {
+		return nil
+	}
+	return cl
+}
+
+func NewRestClientConfig(inClusterFn func() (*rest.Config, error)) (*rest.Config, error) {
+
+	if restConfig, err := inClusterFn(); err == nil {
+		return restConfig, nil
+	}
+
+	pathOptions := clientcmd.NewDefaultPathOptions()
+
+	config, err := pathOptions.GetStartingConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	configOverrides := clientcmd.ConfigOverrides{}
+
+	clientConfig := clientcmd.NewDefaultClientConfig(*config, &configOverrides)
+	restConfig, err := clientConfig.ClientConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return restConfig, nil
+}
+
+// GetApiResources discovers the api-resources for a cluster
+func (cl *DiscoveryClient) GetApiResources() error {
+
+	resourcelist, err := cl.DiscoveryClient.ServerPreferredResources()
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return err
+		}
+		if apierrors.IsForbidden(err) {
+			klog.V(2).Infof("Failed to list objects for Name discovery. Permission denied! Please check if you have the proper authorization")
+		}
+	}
+
+	gvrs := []schema.GroupVersionResource{}
+	for _, rl := range resourcelist {
+		for i := range rl.APIResources {
+			gv, _ := schema.ParseGroupVersion(rl.GroupVersion)
+			ResourceName := rl.APIResources[i].Name
+			g := schema.GroupVersionResource{Group: gv.Group, Version: gv.Version, Resource: ResourceName}
+			gvrs = append(gvrs, g)
+		}
+	}
+
+	var results []map[string]interface{}
+	for _, g := range gvrs {
+		ri := cl.ClientSet.Resource(g)
+		klog.V(2).Infof("Retrieving : %s.%s.%s", g.Resource, g.Version, g.Group)
+		rs, err := ri.List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			klog.V(2).Infof("Failed to retrieve: %s: %s", g, err)
+			continue
+		}
+
+		if len(rs.Items) == 0 {
+			klog.V(2).Infof("No annotations for ResourceVer %s", rs.GetAPIVersion())
+			obj := rs.UnstructuredContent()
+			data, err := json.Marshal(obj)
+			if err != nil {
+				klog.V(2).Infof(err.Error())
+				return nil
+			}
+			output, err := cl.Instance.IsVersioned(data)
+			if err != nil {
+				return err
+			}
+			if output == nil {
+				continue
+			}
+			cl.Instance.Outputs = append(cl.Instance.Outputs, output...)
+
+		} else {
+			for _, r := range rs.Items {
+				if jsonManifest, ok := r.GetAnnotations()["kubectl.kubernetes.io/last-applied-configuration"]; ok {
+					var manifest map[string]interface{}
+
+					err := json.Unmarshal([]byte(jsonManifest), &manifest)
+					if err != nil {
+						klog.V(2).Infof("failed to parse 'last-applied-configuration' annotation of resource %s/%s: %v", r.GetNamespace(), r.GetName(), err)
```

Comment:
Nit: Error message can be handled as a string

```suggestion
						klog.V(2).Infof("failed to parse 'last-applied-configuration' annotation of resource %s/%s: %s", r.GetNamespace(), r.GetName(), err.Error())
```

Reply:
Corrected.

Reply:
HI @sudermanjr , the comments have been taken care off in the latest diffs.